lib/Dialect/OM/Transforms/LinkModules.cpp

```diff
@@ -273,6 +416,18 @@ void LinkModulesPass::runOnOperation() {
       }
   }
 
+  for (auto &[name, symOps] : symbolToOps) {
+
+    if (symOps.size() > 1)
```

Comment:
IMO no need for this check--if the size is zero the look executes zero times.

Reply:
This check avoids the loop when the size is one, that is  only one op with the symbol name.